src/renderers/dom/client/syntheticEvents/SyntheticEvent.js

```diff
@@ -90,17 +100,21 @@ function SyntheticEvent(dispatchConfig, targetInst, nativeEvent, nativeEventTarg
 assign(SyntheticEvent.prototype, {
 
   preventDefault: function() {
+    if (this._destructored) {
+      if (__DEV__) {
+        warning(
+          event,
+          'This synthetic event is reused for performance reasons. If you\'re ' +
+          'seeing this, you\'re calling `preventDefault` on a ' +
+          'released/nullified synthetic event. This is a no-op. See ' +
+          'https://fb.me/react-event-pooling for more information.'
+        );
+      }
+      return;
```

Comment:
Placing `return` inside `__DEV__` means the library can behave differently in development and production, depending on the code below. This can lead to hard-to-reproduce bugs. It’s fine to only log a warning in dev, but all other behavior, including early exits, should be identical.